src/rules.rs

```diff
@@ -237,6 +239,68 @@ pub fn get_recommended_rules() -> Vec<Box<dyn LintRule>> {
     .collect()
 }
 
+/// Returns a list of rules after filtering.
+///
+/// Following rules are applied (in the described order):
+///
+/// - if `maybe_tags` is `None` then all defined rules are returned, otherwise
+///   only rules matching at least one tag will be returned; if provided list
+///   is empty then all rules will be excluded by default
+///
+/// - if `maybe_exclude` is `Some`, all rules with matching codes will
+///   be filtered out
+///
+/// - if `maybe_include` is `Some`, rules with matching codes will be added
+///   to the return list
+///
+/// Before returning the list will be deduplicated and sorted alphabetically.
+pub fn get_filtered_rules(
+  maybe_tags: Option<Vec<String>>,
+  maybe_exclude: Option<Vec<String>>,
+  maybe_include: Option<Vec<String>>,
+) -> Vec<Box<dyn LintRule>> {
+  let mut rules = get_all_rules();
+
+  if let Some(tags) = maybe_tags {
+    let tags_set: HashSet<String> = HashSet::from_iter(tags.into_iter());
+    rules = rules
+      .into_iter()
+      .filter(|rule| {
+        let rule_tags = rule.tags().into_iter().map(|t| t.to_string());
+        let rule_tag_set: HashSet<String> = HashSet::from_iter(rule_tags);
+        !tags_set
+          .intersection(&rule_tag_set)
+          .collect::<Vec<_>>()
+          .is_empty()
```

Comment:
I think you could just do something along the lines of:

```rust
rule.tags().iter().any(|t| tags_set.contains(t))
```

Reply:
I changed it, but handling `String` and `&'static str` is frustrating...